lib/aliquot-pay.rb

```diff
@@ -72,36 +72,63 @@ def self.four_byte_length(str)
     [str.length].pack('V')
   end
 
+  def self.generate_signature(*args)
+    args.map do |s|
+      four_byte_length(s) + s
+    end.join('')
+  end
+
   def self.signature_string(
     message,
-    recipient_id     = DEFAULTS[:merchant_id],
-    sender_id        = DEFAULTS[:info],
-    protocol_version = 'ECv1'
+    recipient_id:     "merchant:#{DEFAULTS[:merchant_id]}",
```

Comment:
Uhm, but if the `recipient_id` does always start with `"merchant:"` then ... couldn't we "hide" this "deeper" so the user calling `.signature_string` will not have to prepend the merchant ID with `"merchant:"`? Perhaps it's nicer to do it like you do — for debugging and spec'ing purposes?

Reply:
If convenient for debugging/spec'ing, you could allow both 🙂